### PR TITLE
Add support for multi party video chat in iOS and Android

### PR DESCRIPTION
**Pull request by lh-rmyers**
This has been tested locally, and it is a bit rough around the edges, but it does work. The rough parts are mainly UI related, where it's not super flashy, but the UI part is more of an example to demonstrate the library working anyway.

### CODE REVIEW

**Comment by AndreiMisiukevich** [comment]
@lh-rmyers amazing! I will review this PR

Have you tested on both platforms?

**Comment by lh-rmyers** [comment]
I have tested on both iOS and Android with real devices. Believe it or not, the problem I was having on Monday was because the height of the video view was set to zero by default when in the StackLayout. So much time for something so simple. I'm totally open to suggestions on how to make this a better pull request, as it's some of my first real work with Xamarin.Forms libraries, so let me know if there's better ways to do certain things.

File: Lib/Xamarin.Forms.OpenTok.iOS/Service/PlatformOpenTokService.cs [hunk 1]
```diff
@@ -140,15 +162,25 @@ private void OnPropertyChanged(object sender, PropertyChangedEventArgs e)
...
+                        //TODO: This is probably wrong.
+                        foreach (OTSubscriber subscriberKit in Subscribers.Select(subscriber => subscriber.Value))
+                        {
+                            subscriberKit.SubscribeToVideo = IsVideoSubscriptionEnabled;
+                        }
+
                         return;
                     case nameof(IsAudioSubscriptionEnabled):
-                        SubscriberKit.SubscribeToAudio = IsAudioSubscriptionEnabled;
+                        //TODO: This is probably wrong.
```
Comment (AndreiMisiukevich) [inline Lib/Xamarin.Forms.OpenTok.iOS/Service/PlatformOpenTokService.cs:178]: Hm, doesn't it work?

File: Lib/Xamarin.Forms.OpenTok/Service/BaseOpenTokService.cs [hunk 2]
```diff
@@ -14,6 +14,8 @@ public abstract class BaseOpenTokService : IOpenTokService
 
         public event Action<string> MessageReceived;
 
+        public event Action<string> SubscriberAdded;
```
Comment (AndreiMisiukevich) [inline Lib/Xamarin.Forms.OpenTok/Service/BaseOpenTokService.cs:17]: What do you think, if we remove this event and add property ObservableCollection<string> or ObservableCollection<OpenTokSubscriberInfo> ?

```public ObservableCollection<OpenTokSubscriberInfo> SubscribersInfo { get; } = new ...```

File: Lib/Xamarin.Forms.OpenTok/OpenTokSubscriberView.cs [hunk 3]
```diff
@@ -2,5 +2,10 @@
 {
     public sealed class OpenTokSubscriberView : OpenTokView
     {
+        public string StreamId { get; set; }
+        public OpenTokSubscriberView(string streamId)
+        {
+            StreamId = streamId;
+        }
```
Comment (AndreiMisiukevich) [inline Lib/Xamarin.Forms.OpenTok/OpenTokSubscriberView.cs:9]: hmmm, how about changing it to a bindable property ?)

**Comment by AndreiMisiukevich** [comment]
wow) that's really great work) i need a lot of time for reviewing
I think it will merge it anyway and then refactor some things
Let's make some changes now and maybe some changes after

File: Lib/Xamarin.Forms.OpenTok.iOS/Service/PlatformOpenTokService.cs (hunk 1 above)
Comment (lh-rmyers) [reply Lib/Xamarin.Forms.OpenTok.iOS/Service/PlatformOpenTokService.cs:178]: To be honest, I never saw the code run. I tried setting a break point on it, but it never got executed. I just assumed at the first point of implementation that iterating over each one to subscribe to each one might not be the right methodology. If it is though, I can remove the TODO.

File: Lib/Xamarin.Forms.OpenTok/Service/BaseOpenTokService.cs (hunk 2 above)
Comment (lh-rmyers) [reply Lib/Xamarin.Forms.OpenTok/Service/BaseOpenTokService.cs:17]: I'm totally open to making it an observable collection. Like I said, I'm still new to this, so I'm still figuring out what the best way to handle new situations is.

File: Lib/Xamarin.Forms.OpenTok/OpenTokSubscriberView.cs (hunk 3 above)
Comment (lh-rmyers) [reply Lib/Xamarin.Forms.OpenTok/OpenTokSubscriberView.cs:9]: Sounds good to me.

File: Lib/Xamarin.Forms.OpenTok.iOS/Service/PlatformOpenTokService.cs (hunk 1 above)
Comment (AndreiMisiukevich) [reply Lib/Xamarin.Forms.OpenTok.iOS/Service/PlatformOpenTokService.cs:178]: You should set IsAudioSubscriptionEnabled to false for hitting this breakpoint, I believe

**Comment by lh-rmyers** [comment]
Do you by any chance have the required code for Xamarin.OpenTok.Android and Xamarin.OpenTok.iOS? I'm having trouble at the moment, and was looking to dig deep into debugging, as well as upgrading those libraries to the latest version. I'd like to generate my own bindings ideally.

**Comment by AndreiMisiukevich** [comment]
> Do you by any chance have the required code for Xamarin.OpenTok.Android and Xamarin.OpenTok.iOS? I'm having trouble at the moment, and was looking to dig deep into debugging, as well as upgrading those libraries to the latest version. I'd like to generate my own bindings ideally.

https://github.com/DreamTeamMobile/Xamarin.Opentok.Samples

**Comment by lh-rmyers** [comment]
> 
> 
> > Do you by any chance have the required code for Xamarin.OpenTok.Android and Xamarin.OpenTok.iOS? I'm having trouble at the moment, and was looking to dig deep into debugging, as well as upgrading those libraries to the latest version. I'd like to generate my own bindings ideally.
> 
> https://github.com/DreamTeamMobile/Xamarin.Opentok.Samples

Unfortunately, that package also just references the Xamarin.OpenTok.Android package instead of being the source for it. It doesn't allow me to build the dll myself.

**Comment by AndreiMisiukevich** [comment]
Great, I will merge your PR on this week, refactor and deploy new package to nuget )

**Comment by AndreiMisiukevich** [comment]
@lh-rmyers merging it for now, then I will refactor it a lot)

**Pull request merged**